File: puremvc/puremvc/MVC/Patterns/Observer/Notifier.cs
using System;
using MVC.Interfaces;

namespace MVC.Patterns.Observer
{
    public class Notifier: INotifier
    {
        public virtual void SendNotification(string notificationName, object body = null, string type = null)
        {
            Facade.SendNotification(notificationName, body, type);
        }

        public void InitializeNotifier(string key)
        {
            MultitonKey = key;
        }

        protected IFacade Facade
        {
            get{
                if (MultitonKey == null) throw new Exception(MULTITON_MSG);
                return Patterns.Facade.Facade.GetInstance(MultitonKey, key => new Facade.Facade(key));
            }
        }

        public string MultitonKey {get; protected set;}

        protected string MULTITON_MSG = "multitonKey for this Notifier not yet initialized";
    }
}

File: framework/project/base/XScripts/Frameworks/MVC/Patterns/Observer/Notifier.lua
local Notifier = Class("MVC.Notifier",{})

function Notifier:Ctor() end

function Notifier:SendNotification(notificationName, body, type) 
    local facade = self:GetFacade()
    if facade ~= nil then
        facade.SendNotification(notificationName, body, type)
    end
end
function Notifier:InitializeNotifier(key) 
    self.multitonKey = key
    self.facade = self:GetFacade()
end

function Notifier:GetFacade()
    if self.multitonKey == nil then
        error(Notifier.MULTITON_MSG)
    end
    return MVC.Facade.GetInstance(self.multitonKey)
end

Notifier.MULTITON_MSG = "multitonKey for this Notifier not yet initialized!"
return Notifier